### DISCUSSION TITLE
How to add  '-std=c++17'  to compiler flags

### DISCUSSION THREAD

**Discussion by Bigdogcat** [idaholab/moose#24957] [Q&A General] [answered]
I'd like to ask how to add  '-std=c++17'  to compiler flags. I'm asking this because after updating MOOSE, the following error comes up when compiling

/Users/sunqingpei/projects/moose/framework/build/header_symlinks/Moose.h:55:1: error: static_assert failed due to requirement '201402L >= 201703L' "MOOSE requires a C++17 compatible compiler (GCC >= 7.5.0, Clang >= 5.0.2). Please update your compiler or, if compatible, add '-std=c++17' to your compiler flags and try again. If using the MOOSE conda package, please attempt a MOOSE environment update (using `mamba update --all`). If this update is not successful, please create a new MOOSE environment (see https://mooseframework.inl.gov/getting_started/installation/conda.html#uninstall-conda-moose-environment)."

**Comment by GiudGiud** [comment, marked as answer]
You can type ‘env ‘ and look at all the environment variables defined, in particular the compiling flags. 

You can then add this flags to CXXFLAGS using export CXXFLAGS="—std=c++17:$CXXFLAGS"

**Reply by GiudGiud** [reply]
You should not have opened two posts for the same issue. Closing this one
